mobile/src/main/AndroidManifest.xml

```diff
@@ -2,6 +2,10 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="ch.epfl.sdp.mobile">
 
+    <!-- Necessary permissions for speech recognition -->
+    <uses-permission android:name="android.permission.RECORD_AUDIO"/>
+    <uses-permission android:name="android.permission.INTERNET"/>
```

Comment:
The `android.permission.INTERNET` is needed for retrieving remote results. It increases the accuracy of the recognizer. Removing it will make the recognizer rely solely on a limited local language model.

Reply:
Alright, sounds good. I thought that because `SpeechRecognizer` is a system service, the details of whether it's using a recogniser with / without internet are internal to the framework and that clients therefore don't need the `android.permission.INTERNET` because it would be the responsibility of the speech recognition service to actually access internet.

Do you have by any change some documentation on this ?

Reply:
As discussed in private we concluded that there was no need to grant such permission.